torch/csrc/jit/passes/quantization.cpp

```diff
@@ -572,5 +599,107 @@ template TORCH_API void InsertQuantDequantNodesForParam(
         getQParamFunc,
     at::ScalarType t);
 
+
+static Node* prepQuantAddObserverFor(
+    Value* v,
+    Node* original_observer_node,
+    Node* insert_point) {
+  TORCH_INTERNAL_ASSERT(insert_point != nullptr);
+  WithInsertPoint ins(insert_point);
+
+  // Create a new observer node. We just need to clone the original one.
+  Node* observerNode = insert_point->owningGraph()->createClone(
+      &*original_observer_node, [&](Value* v) { return v; }, false);
+
+  // Set the type and the name of the output of the new observer node. It will
+  // be used instead of the original value v.
+  Value* observedValue = observerNode->addOutput();
+  observedValue->setType(v->type());
+  observedValue->setDebugName(v->debugName() + ".observed");
+
+  // Now we can add the inputs.
+  observerNode->addInput(v);
+  return observerNode;
+}
+
+TORCH_API script::Module PrepareQuant(
+    const script::Module& module,
+    const std::string& method_name,
+    const script::Module& observer_module,
+    const script::Module& weight_observer_module) {
+  script::Module input_module = module.clone();
+  script::Method method = input_module.get_method(method_name);
+  auto graph = method.graph();
+  auto num_activation_inputs = method.num_inputs();
+  TORCH_CHECK(graph != nullptr);
+  // num_activation_inputs is the number of activations or external data
+  // excluding the parameters
+  TORCH_CHECK(num_activation_inputs <= graph->inputs().size());
```

Comment:
This check is useless since `method.num_inputs()` and `method.graph().num_inputs()` are the same. That said, the first argument of a method is `%self`, which probably doesn't need to be observed.

Reply:
> That said, the first argument of a method is %self, which probably doesn't need to be observed.
What do you mean?

Reply:
The first argument of a method is always `%self` (the parent module). It is never a tensor so we never observe it. The check below handles it, but that check will always result in "do not observe" for the first argument, so we can just start from the second argument. I don't know if it's worth changing though.

Reply:
oh, sure, I can change that